Lib/asyncio/locks.py

```diff
@@ -417,6 +424,7 @@ def _wake_up_next(self):
             if not fut.done():
                 self._value -= 1
                 fut.set_result(True)
+                # assert fut.true() and not fut.cancelled()
```

Comment:
I'd remove that -- I don't think there's such a thing as `fut.true()` and the semantics of `set_result()` ensure that if the future was cancelled it will raise an exception.

Reply:
(Ditto if it already had another result, or an exception.)